client/signals/handlers.py

```diff
@@ -0,0 +1,134 @@
+"""
+Contains signals, responsible to sync edX with NodeBB.
+As some related event is occurred in edX the signal is received
+and a relevant request is made to NodeBB write api to make that
+change at NodeBB side too.
+"""
+from django.contrib.auth.models import User
+from django.db.models.signals import post_save, pre_delete
+from django.dispatch import receiver
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.features.openedx_nodebb_discussion.client.tasks import (
+    task_create_user_on_nodebb, task_sync_user_profile_info_with_nodebb,
+    task_delete_category_from_nodebb, task_delete_user_from_nodebb,
+    task_create_category_on_nodebb, task_join_group_on_nodebb,
+    task_unjoin_group_on_nodebb
+)
+from openedx.features.openedx_nodebb_discussion.models import EdxNodeBBCategory
+from student.models import UserProfile, CourseEnrollment
+
+
+@receiver(post_save, sender=User)
+def create_and_update_user_on_nodebb(sender, instance, created, update_fields, **kwargs):
```

Comment:
Create and update should be separate functions. Don't copy PhilU

Reply:
I don't think it can be separated. Because whenever user information is changed or a new user created this receiver will receive the signal. We are separating the task by using if condition below.